src/onemod/fsutils/config_loader.py

```diff
@@ -0,0 +1,28 @@
+from pathlib import Path
+from typing import Any
+
+from onemod.fsutils.io import ConfigIO, configio_dict
+
+
+class ConfigLoader:
+    """Handles loading and dumping of configuration files and serialized models."""
+
+    io_dict: dict[str, ConfigIO] = configio_dict
+
+    def load(self, path: Path, **options) -> Any:
+        """Load a config file or serialized model."""
+        if path.suffix not in self.io_dict:
+            raise ValueError(f"Unsupported config format for '{path.suffix}'")
+
+        return self.io_dict[path.suffix].load(path, **options)
+
+    def dump(self, obj: Any, path: Path, **options) -> None:
+        """Save a config or model object."""
+        if path.suffix not in self.io_dict:
+            raise ValueError(f"Unsupported config format for '{path.suffix}'")
+
+        self.io_dict[path.suffix].dump(obj, path, **options)
+
+    def __repr__(self) -> str:
+        base_repr = super().__repr__()
+        return f"{base_repr[:-1]}, handles_configs=True\n)"
```

Comment:
Since `ConfigLoader` is not subclass for anything, `base_repr` will be something like `<...>` I don't know if it will play well with the additional `handles_configs` appended at the end.

Reply:
Ah, yup I did not update `__repr__()` after making certain design changes thanks 👍